core/arch/arm/plat-ls/main.c

```diff
@@ -161,3 +164,41 @@ void main_secondary_init_gic(void)
 {
 	gic_cpu_init(&gic_data);
 }
+
+#ifdef CFG_HW_UNQ_KEY_REQUEST
+
+#ifndef ASM
+#include <types_ext.h>
+int get_hw_unique_key(uint64_t smc_func_id, uint64_t in_key, uint64_t size);
+#endif
+
+/*
+ * Issued when requesting to Secure Storage Key for secure storage.
+ *
+ * SiP Service Calls
+ *
+ * Register usage:
+ * r0/x0	SMC Function ID, OPTEE_SMC_FUNCID_SIP_LS_HW_UNQ_KEY
+ */
+#define OPTEE_SMC_FUNCID_SIP_LS_HW_UNQ_KEY			0xFF14
+#define OPTEE_SMC_FAST_CALL_SIP_LS_HW_UNQ_KEY \
+	OPTEE_SMC_CALL_VAL(OPTEE_SMC_32, OPTEE_SMC_FAST_CALL, \
+			   OPTEE_SMC_OWNER_SIP, \
+			   OPTEE_SMC_FUNCID_SIP_LS_HW_UNQ_KEY)
+
+void tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
+{
+	int ret = 0;
+
+	ret = get_hw_unique_key(OPTEE_SMC_FAST_CALL_SIP_LS_HW_UNQ_KEY,
+			virt_to_phys(&hwkey->data[0]), sizeof(hwkey->data));
+
+	/* Invalidate cache to fetch data from external memory */
+	cache_op_inner(DCACHE_AREA_INVALIDATE,
+			&hwkey->data[0], sizeof(hwkey->data));
+
+	if (ret < 0)
+		EMSG("\nCould not able to fetch the HW Unique Key.\
+			\nContinuing with dummy values.\n");
```

Comment:
- Why use a line continuation here?  It will needlessly add spaces to the output.
- Also, the terminating `\n` is not needed (`EMSG()` adds it for you)
- `s/able to //`

```C
		EMSG("\nCould not fetch the HW Unique Key."
		     "\nContinuing with dummy values.");
```

Reply:
Now. Made a single line comment.